internal/jujuapi/streamcontrollerproxy.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 Canonical.
+
+package jujuapi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/auth"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/jimmhttp"
+	"github.com/canonical/jimm/v3/internal/streamproxy"
+)
+
+const (
+	logTransferPath = "/migrate/logtransfer"
+)
+
+// streamControllerProxier serves any HTTP endpoints
+// served against a controller's root address rather than
+// a model specific path segment i.e. /model/{uuid}/*.
+// Messages are handled by proxying them between the
+// controller and client.
+type streamControllerProxier struct {
+	apiServer
```

Comment:
We're seemingly only using jimm on the API server here, and we've implemented ServeWS, do we need an apiServer or just a JIMM?